### PR TITLE
drivers: flash: Fix flash_write_protection_set usage

### PR DESCRIPTION
**Pull request by eugmes**
This PR cleans up usage of `flash_write_protection_set()`. The second argument of this function as `bool` so passing 0 and 1 is not entirely correct. The PR also changes `flash_shell.c` so that flash protection is disabled before each flash erase or write operation because drivers may automatically enable flash protection once such operation is complete.

### CODE REVIEW

**Review by Laczen — APPROVED** [review]
Thanks for the improvement.

**Review by nvlsianpu — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

File: drivers/flash/flash_shell.c [hunk 1]
```diff
@@ -193,13 +193,13 @@ static int cmd_test(const struct shell *shell, size_t argc, char *argv[])
...
 	}
 
-	flash_write_protection_set(flash_dev, 0);
-
 	for (uint32_t i = 0; i < size; i++) {
 		test_arr[i] = (uint8_t)i;
 	}
 
 	while (repeat--) {
+		flash_write_protection_set(flash_dev, false);
```
Comment (ioannisg) [inline drivers/flash/flash_shell.c:201]: Just wondering: shouldn't write protection be "restored" before (or right after) exiting the while loop?

@nvlsianpu ^^

Comment (nvlsianpu) [reply drivers/flash/flash_shell.c:201]: In that case should be restored after while loop.

In general recommendation is to set flash protecting after packet of write/erase operation was completed.

Comment (nvlsianpu) [reply drivers/flash/flash_shell.c:201]: @eugmes ^^

Comment (eugmes) [reply drivers/flash/flash_shell.c:201]: @nvlsianpu I can fix this, but the original code also didn't restore write protection.

Comment (eugmes) [reply drivers/flash/flash_shell.c:203]: Fixed. It's either `goto` or duplicate code.

Comment (nvlsianpu) [reply drivers/flash/flash_shell.c:203]: thanks!

Comment (ioannisg) [reply drivers/flash/flash_shell.c:203]: wonder if we really need to have "goto" here, isn't there a refactor that removes the need, without duplicating code too much?

Comment (eugmes) [reply drivers/flash/flash_shell.c:205]: @ioannisg How about this?

Comment (eugmes) [reply drivers/flash/flash_shell.c:205]: oops.

Comment (eugmes) [reply drivers/flash/flash_shell.c:205]: Now.

**Review by ioannisg — APPROVED** [review]
LGTM
@nvlsianpu , maybe another quick look?

**Pull request merged**